core/chains/evm/logpoller/orm.go

```diff
@@ -122,7 +122,7 @@ func (o *ORM) SelectLatestLogEventSigWithConfs(eventSig common.Hash, address com
          WHERE evm_chain_id = $1 
             AND event_sig = $2 
             AND address = $3 
-            AND (block_number + $4) <= (SELECT COALESCE(block_number, 0) FROM evm.log_poller_blocks WHERE evm_chain_id = $1 ORDER BY block_number DESC LIMIT 1)
+            AND block_number <= (SELECT COALESCE(block_number, 0) FROM evm.log_poller_blocks WHERE evm_chain_id = $1 ORDER BY block_number DESC LIMIT 1) - $4
```

Comment:
Careful! This was set up deliberately for optimization reasons IIRC

Reply:
I was the one who did these optimizations :P For some reason, I missed this query, therefore fixing it right now

Reply:
https://github.com/smartcontractkit/chainlink/pull/9968